vignettes/checks/sourceConceptRecordCompleteness.Rmd

```diff
@@ -23,24 +23,50 @@ The number and percent of records with a value of 0 in the source concept field
 
 ## Definition
 
-- *Numerator*: 
-- *Denominator*:
-- *Related CDM Convention(s)*:
-- *CDM Fields/Tables*:
-- *Default Threshold Value*:
+- *Numerator*: The number of rows with a value of 0 in the `X_source_concept_id` source concept field.  In the case of `MEASUREMENT.unit_source_concept_id` and `OBSERVATION.unit_source_concept_id`, the number of rows with a value of 0 in the `X_source_concept_id` source concept field AND a non-NULL `value_as_number`. 
+- *Denominator*: The total number of rows in the table. In the case of `MEASUREMENT.unit_source_concept_id` and `OBSERVATION.unit_source_concept_id`, the number of rows with non-NULL `value_as_number`. 
```

Comment:
It might be because it is late, but I can't find `OBSERVATION.unit_source_concept_id`. Only Measurement and Device Exposure have a unit_source_concept_id.

Reply:
whoops, that was my error - will update to reference device instead of observation here

Reply:
I see what happened, we use the same query for both standard and source concept id, with the exception for unit concepts.
https://github.com/OHDSI/DataQualityDashboard/blob/6ef7ee2dd1116741e3fe9907ef4d9cc98eccb96c/inst/sql/sql_server/field_concept_record_completeness.sql#L37

Note that we don't have this exception for the device table (`unit_concept_id` nor `unit_source_concept_id`) as it does not have a `value_as_number`. I assume it refers to the `quantity` field. Same for specimen table.

So we actually need to update the query to be consistent. It becomes a bit messy, but this would be the new where-clause.
```sql
WHERE cdmTable.@cdmFieldName = 0 
{@cdmTableName IN ('MEASUREMENT', 'OBSERVATION') & @cdmFieldName IN ('UNIT_CONCEPT_ID', 'UNIT_SOURCE_CONCEPT_ID')}?{AND cdmTable.value_as_number IS NOT NULL}
{@cdmTableName IN ('DEVICE_EXPOSURE', 'SPECIMEN') & @cdmFieldName IN ('UNIT_CONCEPT_ID', 'UNIT_SOURCE_CONCEPT_ID')}?{AND cdmTable.quantity IS NOT NULL}
```
That said, after thinking this through for a bit, if the ETL is properly defined it should not be an issue (i.e. if value is empty, then also unit should be empty)

Reply:
Oh my!  Great catch; we don't even check this on source concept ID fields, or any field in device or specimen.  So for now I will update this to remove the language altogether for source concept ID.  I'll file an issue to get this updated.

Reply:
@katy-sadowski Opened the issue here: https://github.com/OHDSI/DataQualityDashboard/issues/558

Reply:
Thank you!!!